src/dryad/LegendreAngularDistribution/test/LegendreAngularDistribution.test.cpp

```diff
@@ -0,0 +1,350 @@
+// include Catch2
+#include <catch2/catch_test_macros.hpp>
+#include <catch2/matchers/catch_matchers_floating_point.hpp>
+using Catch::Matchers::WithinRel;
+using Catch::Matchers::WithinAbs;
+
+// what we are testing
+#include "dryad/LegendreAngularDistribution.hpp"
+
+// other includes
+
+// convenience typedefs
+using namespace njoy::dryad;
+
+SCENARIO( "LegendreAngularDistribution" ) {
+
+  GIVEN( "Legendre coefficients for an expansion" ) {
+
+    WHEN( "the data is given explicitly" ) {
+
+      std::vector< double > coefficients = { -31./3., 73./5., -14./3., 2./5. };
+
+      LegendreAngularDistribution chunk( std::move( coefficients ) );
+
+      THEN( "a LegendreAngularDistribution can be constructed and members can "
+            "be tested" ) {
+
+        CHECK_THAT( -1., WithinRel( chunk.lowerCosineLimit() ) );
+        CHECK_THAT(  1., WithinRel( chunk.upperCosineLimit() ) );
+
+        CHECK( 3 == chunk.order() );
+
+        CHECK( 4 == chunk.coefficients().size() );
+        CHECK_THAT( -10.33333333333333, WithinRel( chunk.coefficients()[0] ) );
+        CHECK_THAT(  14.6             , WithinRel( chunk.coefficients()[1] ) );
+        CHECK_THAT(  -4.66666666666666, WithinRel( chunk.coefficients()[2] ) );
+        CHECK_THAT(   0.4             , WithinRel( chunk.coefficients()[3] ) );
+      } // THEN
+
+      THEN( "a LegendreAngularDistribution can be evaluated" ) {
+
+        CHECK_THAT(  -8., WithinRel( chunk(  0. ) ) );
+        CHECK_THAT(   0., WithinAbs( chunk(  1. ), 1e-12 ) );
+        CHECK_THAT( -30., WithinRel( chunk( -1. ) ) );
+      } // THEN
+
+      THEN( "a LegendreAngularDistribution can be linearised" ) {
+
+        ToleranceConvergence convergence( 0.01 );
+        TabulatedAngularDistribution linear = chunk.linearise( convergence );
+
+        CHECK( 21 == linear.numberPoints() );
+        CHECK( 1 == linear.numberRegions() );
+
+        CHECK( 21 == linear.cosines().size() );
+        CHECK( 21 == linear.values().size() );
+        CHECK( 1 == linear.boundaries().size() );
+        CHECK( 1 == linear.interpolants().size() );
+
+        CHECK( 20 == linear.boundaries()[0] );
+
+        CHECK( InterpolationType::LinearLinear == linear.interpolants()[0] );
+
+        CHECK_THAT( -1.0      , WithinRel( linear.cosines()[0] ) );
```

Comment:
I actually went and bothered to verify this linearization. It's below 0.01 relative error, except between the last two nodes. The midpoint is below 0.01, but of course it diverges to infinity as it approaches zero. So, this works as expected.